pkg/streamingpromql/operators/subquery.go

```diff
@@ -130,8 +167,24 @@ func (s *Subquery) ExpressionPosition() posrange.PositionRange {
 	return s.expressionPosition
 }
 
+func (s *Subquery) Prepare(ctx context.Context, params *types.PrepareParams) error {
+	s.parentQueryStats = params.QueryStats
+	// Final aggregation into parent stats happens in samplesProcessedInSubquery().
+	// Child samples always counted per step to count parent stats correctly.
+	childStats, err := types.NewQueryStats(s.SubqueryTimeRange, true, s.memoryConsumptionTracker)
+	if err != nil {
+		return fmt.Errorf("could not create stats for subquery: %w", err)
```

Comment:
We shouldn't wrap this error, for the same reasons as elsewhere: the only error `NewQueryStats` can return is a limit exceeded error, and so we want to return that as-is to users.

Reply:
pushed the fix